--- gnome-shell-3.38.3.orig/js/ui/screenShield.js
+++ gnome-shell-3.38.3/js/ui/screenShield.js
@@ -26,6 +26,12 @@ const DISABLE_LOCK_KEY = 'disable-lock-s
 
 const LOCKED_STATE_STR = 'screenShield.locked';
 
+const LOGINSCREEN_SCHEMA = 'com.ubuntu.login-screen';
+const LOGINSCREEN_BACKGROUND_COLOR_KEY = 'background-color';
+const LOGINSCREEN_BACKGROUND_PICTURE_URI_KEY = 'background-picture-uri';
+const LOGINSCREEN_BACKGROUND_REPEAT_KEY = 'background-repeat';
+const LOGINSCREEN_BACKGROUND_SIZE_KEY = 'background-size';
+
 // ScreenShield animation time
 // - STANDARD_FADE_TIME is used when the session goes idle
 // - MANUAL_FADE_TIME is used for lowering the shield when asked by the user,
@@ -119,6 +125,13 @@ var ScreenShield = class {
         this._lockSettings = new Gio.Settings({ schema_id: LOCKDOWN_SCHEMA });
         this._lockSettings.connect('changed::%s'.format(DISABLE_LOCK_KEY), this._syncInhibitor.bind(this));
 
+        // ...
+        // ...
+        // ...
+        // ...
+        // ...
+        // ...
+
         this._isModal = false;
         this._isGreeter = false;
         this._isActive = false;
@@ -196,6 +209,26 @@ var ScreenShield = class {
         return this._isModal;
     }
 
+    _refreshBackground() {
+        let inline_style = [];
+
+        let backgroundColor = "#FF0000";
+        let backgroundPictureUri = "file:///etc/nixos/background.jpg";
+        let backgroundRepeat = "no-repeat";
+        let backgroundSize = "cover";
+
+        if (backgroundColor != "")
+            inline_style.push("background-color: " + backgroundColor);
+        if (backgroundPictureUri != "")
+            inline_style.push("background-image: url(" + backgroundPictureUri + ")");
+        if (backgroundRepeat != "default")
+            inline_style.push("background-repeat: " + backgroundRepeat);
+        if (backgroundSize != "default")
+            inline_style.push("background-size: " + backgroundSize);
+
+        this._lockDialogGroup.set_style(inline_style.join('; '));
+    }
+
     _syncInhibitor() {
         let lockEnabled = this._settings.get_boolean(LOCK_ENABLED_KEY) ||
                           this._settings.get_boolean(SUSPEND_LOCK_ENABLED_KEY);
